File: src/main/java/br/com/dev/ecommerce/estoque/service/ProdutoService.java
package br.com.dev.ecommerce.estoque.service;

import java.util.List;

import br.com.dev.ecommerce.estoque.dto.ProdutoDTO;
import br.com.dev.ecommerce.estoque.dto.ProdutoDetalheDTO;

public interface ProdutoService {
	
	public ProdutoDetalheDTO buscar(Long id);		
	
	public void salvar(ProdutoDTO dto);
	
	public void excluir(Long id);
	
	public void atualizar(Long id, ProdutoDTO dto);
	
	public List<ProdutoDetalheDTO> getProdutos();

}

File: frontend/src/app/estoque/produto/produto.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ProdutoService {

  private readonly PATH = "/api/produto";

  constructor(
    private http: HttpClient
  ) { }

  buscar(id: number): Observable<any> {
    return this.http.get(`${this.PATH}/buscar/${id}`);
  }

  salvar(produto: any): Observable<any> {
    return this.http.post(`${this.PATH}/salvar`, produto);
  }

  excluir(id: number): Observable<any> {
    return this.http.delete(`${this.PATH}/excluir/${id}`);
  }

  atualizar(id: number, novoProduto: any): Observable<any> {
    return this.http.put(`${this.PATH}/atualizar/${id}`, novoProduto);
  }

  getProdutos(): Observable<any> {
    return this.http.get(`${this.PATH}`);
  }

}
